### PR TITLE
HF landscape

### PR DESCRIPTION
**Pull request by mdietze**
Code to doing a landscape-scale SDA, using Harvard Forest as the test case

### CODE REVIEW

File: modules/assim.sequential/R/sda.enkf_MultiSite.R
Comment (helge22a) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:1]: Are you sure this is the most recent version of this function? The check for observations and for the analysis to skip if there are no observations was fixed in one of the last PR but it looks like this is the old version?

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 1]
```diff
@@ -125,9 +125,9 @@ sda.enkf.multisite <- function(settings,
   #Here I'm trying to make a temp config list name and put it into map to iterate
   if(multi.site.flag){
     conf.settings<-settings
-    site.ids <- conf.settings %>% purrr::map(~.x[['run']] ) %>% purrr::map('site') %>% purrr::map('id') %>% unlist() %>% as.character()
+    site.ids <- conf.settings$run %>% map('site') %>% map('id') %>% unlist() %>% as.character()
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:128]: note to self: somehow namespaces got lost in a merge

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 2]
```diff
@@ -344,14 +345,16 @@ sda.enkf.multisite <- function(settings,
       new.params <- sda_matchparam(settings, ensemble.samples, site.ids, nens)
     }
       #sample met ensemble members
-      inputs <- conf.settings %>% purrr::map(function(setting) {
+      #TODO: incorporate Phyllis's restart work
+      #      sample all inputs specified in the settings$ensemble not just met
+      inputs <- papply(conf.settings,function(setting) {
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:350]: add namespace to papply

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 3]
```diff
@@ -365,9 +368,9 @@ sda.enkf.multisite <- function(settings,
       if (t>1){
         #for next time step split the met if model requires
         #-Splitting the input for the models that they don't care about the start and end time of simulations and they run as long as their met file.
-        inputs.split <- PEcAnAssimSequential::metSplit(conf.settings, inputs, settings, model, no_split = FALSE, obs.times, t, nens, restart_flag = FALSE, my.split_inputs)
+        inputs.split <- metSplit(conf.settings, inputs, settings, model, no_split = FALSE, obs.times, t, nens, restart_flag = FALSE, my.split_inputs)
```
Comment (mdietze) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:371]: namespace

File: modules/assim.sequential/R/sda.enkf_MultiSite.R [hunk 4]
```diff
@@ -500,18 +511,10 @@ sda.enkf.multisite <- function(settings,
...
-          recompileTobit = FALSE
-        }
-        
-        if(!exists('Cmcmc')) {
-          recompileGEF = TRUE
-        }else{
-          recompileGEF = FALSE
-        }
+
+        recompileTobit = !exists('Cmcmc_tobit2space')
```
Comment (DongchenZ) [inline modules/assim.sequential/R/sda.enkf_MultiSite.R:515]: Great one!

**Comment by mdietze** [comment]
@DongchenZ I think this code is ready to go. Could you give a quick look and then "approve" if you think its good.

@helge22a as far as I can tell my code doesn't actively revert any of the changes you mentioned, as much as they seem to have gotten lost along the way independent of this PR. Would be good to track those down and get them applied, but I don't think there's anything I can do in this PR to fix that.

**Pull request merged**